leader/clique/async_korach_moran_zaks/async_korach_moran_zaks.go

```diff
@@ -0,0 +1,371 @@
+package async_korach_moran_zaks
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/krzysztof-turowski/distributed-framework/lib"
+)
+
+type state struct {
+	Active      bool
+	Phase       int
+	Rivals      map[int]int
+	Vassals     map[int]int
+	Overlord    int
+	King        int
+	AskedBefore int
+	Queue       []message
+}
+
+type message struct {
+	Flag  int
+	Phase int
+	King  int
+	Index int
+}
+
+const (
+	ASK = iota
+	ACCEPT
+	UPDATE
+	YOUR_CITIZEN
+	I_AM_THE_KING
+)
+
+/************************************************************/
+/*                           KING                           */
+/************************************************************/
+
+func mainKing(node lib.Node) {
+	for {
+		if processKing(node) {
+			break
+		}
+	}
+
+	if !getState(node).Active {
+		mainCitizen(node)
+	} else {
+		iAmTheKing(node)
+	}
+}
+
+func processKing(node lib.Node) bool {
+
+	state := getState(node)
+
+	defer setState(node, &state)
+
+	send := func(index int, message []byte) {
+		node.SendMessage(index, message)
+	}
+
+	//len(state.Rivals) == 0 means that every ASK provoked the response -> we are the King of Kings
+	if !state.Active || len(state.Rivals) == 0 {
+		return true
+	} else {
+		index := getSomeKey(state.Rivals)
+		delete(state.Rivals, index)
+		send(index, createMessage(ASK, state.Phase, node.GetIndex()))
+	}
+
+ASK_RESPONSE:
+	received_index, received_message := receiveMessage(node)
+	message_type := received_message.Flag
+
+	switch message_type {
+	case ASK:
+		if state.Phase > received_message.Phase || (state.Phase == received_message.Phase && state.King > received_message.King) {
+			goto ASK_RESPONSE
```

Comment:
Please re-write it as a for loop from L73 on with continue here

Reply:
It is no longer needed to have goto or for loop here